### ISSUE TITLE
RFC 1214 warning in librustc_mir/repr.rs

### ISSUE THREAD

**Issue by apasel422** [rust-lang/rust#28999]
Building nightly yields
```
../rust/src/librustc_mir/repr.rs:404:5: 404:38 warning: the parameter type `V` may not live long enough [E0309]
../rust/src/librustc_mir/repr.rs:404     pub elem: ProjectionElem<'tcx, V>,
                                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../rust/src/librustc_mir/repr.rs:401:24: 401:33 note: in this expansion of #[derive_PartialEq] (defined in ../rust/src/librustc_mir/repr.rs)
../rust/src/librustc_mir/repr.rs:404:5: 404:38 help: run `rustc --explain E0309` to see a detailed explanation
../rust/src/librustc_mir/repr.rs:404:5: 404:38 help: consider adding an explicit lifetime bound `V: 'tcx`...
../rust/src/librustc_mir/repr.rs:404:5: 404:38 note: this warning results from recent bug fixes and clarifications; it will become a HARD ERROR in the next release. See RFC 1214 for details.
../rust/src/librustc_mir/repr.rs:404     pub elem: ProjectionElem<'tcx, V>,
                                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../rust/src/librustc_mir/repr.rs:401:24: 401:33 note: in this expansion of #[derive_PartialEq] (defined in ../rust/src/librustc_mir/repr.rs)
../rust/src/librustc_mir/repr.rs:404:5: 404:38 note: ...so that the reference type `&repr::ProjectionElem<'_, V>` does not outlive the data it points at
../rust/src/librustc_mir/repr.rs:404     pub elem: ProjectionElem<'tcx, V>,
                                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../rust/src/librustc_mir/repr.rs:401:24: 401:33 note: in this expansion of #[derive_PartialEq] (defined in ../rust/src/librustc_mir/repr.rs)
../rust/src/librustc_mir/repr.rs:404:5: 404:38 warning: the parameter type `V` may not live long enough [E0309]
../rust/src/librustc_mir/repr.rs:404     pub elem: ProjectionElem<'tcx, V>,
                                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../rust/src/librustc_mir/repr.rs:401:24: 401:33 note: in this expansion of #[derive_PartialEq] (defined in ../rust/src/librustc_mir/repr.rs)
../rust/src/librustc_mir/repr.rs:404:5: 404:38 help: run `rustc --explain E0309` to see a detailed explanation
../rust/src/librustc_mir/repr.rs:404:5: 404:38 help: consider adding an explicit lifetime bound `V: 'tcx`...
../rust/src/librustc_mir/repr.rs:404:5: 404:38 note: this warning results from recent bug fixes and clarifications; it will become a HARD ERROR in the next release. See RFC 1214 for details.
../rust/src/librustc_mir/repr.rs:404     pub elem: ProjectionElem<'tcx, V>,
                                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../rust/src/librustc_mir/repr.rs:401:24: 401:33 note: in this expansion of #[derive_PartialEq] (defined in ../rust/src/librustc_mir/repr.rs)
../rust/src/librustc_mir/repr.rs:404:5: 404:38 note: ...so that the reference type `&repr::ProjectionElem<'_, V>` does not outlive the data it points at
../rust/src/librustc_mir/repr.rs:404     pub elem: ProjectionElem<'tcx, V>,
                                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../rust/src/librustc_mir/repr.rs:401:24: 401:33 note: in this expansion of #[derive_PartialEq] (defined in ../rust/src/librustc_mir/repr.rs)
```

**Comment by apasel422** [comment]
A reduction of the code in question is available here: https://play.rust-lang.org/?gist=9893feaa919468e9f534&version=nightly

CC @nikomatsakis @arielb1.

**Comment by arielb1** [comment]
Further reduced:

```Rust
pub struct Xyz<'a, V> {
    v: (V, &'a u32),
}

impl<'a, V: PartialEq> PartialEq for Xyz<'a, V> {
    fn eq(&self, other: &Self) -> bool {
        self.v == other.v
    }
}

fn main() {}
```

**Comment by arielb1** [comment]
It looks like the borrow of `this.v` is not constrained and is therefore inferred to `'a`.
